### PR TITLE
Silk.NET backed Skia

### PR DESCRIPTION
**Pull request by HurricanKai**
Just a basic Silk.NET backed Skia demo on OpenGL. We could consider adding this as a demo, instead of a Lab app.
@Perksey, thoughts?

### CODE REVIEW

**Comment by Perksey** [comment]
Yeah making it an official demo seems fine, will review it accordingly though.

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:6]: Should we start using `with` in our tutorials?

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:11]: Why?

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:12]: Could do with some whitespace before this line for readability

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:13]: Why do we need to use Initialize? We should encourage the use of Load callbacks, this function is only intended for niche use cases.

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:14]: Can you add a comment on what this does.

Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:14]: Why do we need the IntPtr cast?

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:17]: What are these values? Can you add a comment.

Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:17]: Can some of these numbers be grabbed from WindowOptions to prevent duplication (i.e. you only need to change the size in one place)

Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:17]: What happens if you resize the window? These seem like size-dependent values, so do we need to recreate this in a FramebufferResize callback?

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:21]: Use `delta` instead of `d` for explicitness.

File: src/Lab/Experiments/SilkBackedSkiaRendering/SilkBackedSkiaRendering.csproj
Comment (Perksey) [inline src/Lab/Experiments/SilkBackedSkiaRendering/SilkBackedSkiaRendering.csproj:13]: Why are you pulling in just Glfw? Why not the Windowing metapacakge? (would eliminate the .Common reference as well)

**Review by Perksey — CHANGES_REQUESTED** [review]

**Comment by Perksey** [comment]
If you don't want to graduate this from the Lab you can discard all of the above comments, just want to ensure we keep a high standard for our tutorials.

File: src/Lab/Experiments/SilkBackedSkiaRendering/SilkBackedSkiaRendering.csproj
Comment (HurricanKai) [reply src/Lab/Experiments/SilkBackedSkiaRendering/SilkBackedSkiaRendering.csproj:13]: Because this only works with GLFW. Will see if I can also make it compatible with SDL

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (HurricanKai) [reply src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:21]: Is _ also fine?

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (HurricanKai) [reply src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:17]: Yes. Will do

File: src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs
Comment (Perksey) [reply src/Lab/Experiments/SilkBackedSkiaRendering/Program.cs:21]: I mean yes, but it might be unclear what we're discarding especially given this is a minimal example. So probably worth naming the variable anyway.

File: src/Lab/Experiments/SilkBackedSkiaRendering/SilkBackedSkiaRendering.csproj
Comment (Perksey) [reply src/Lab/Experiments/SilkBackedSkiaRendering/SilkBackedSkiaRendering.csproj:13]: There's no reason that I know why this wouldn't work with either, so give it a try :) if it doesn't work with both we should probably keep this in the Lab.

**Review by Perksey — CHANGES_REQUESTED** [review]
Not sure if you intended to request another review, but my previous comments still stand.

**Comment by HurricanKai** [comment]
I have no idea why, but it automatically requested a review from the codeowners :thinking:

**Comment by Perksey** [comment]
Oh just noticed you moved this into the lab. Happy to merge once conflicts are resolved.

**Review by Perksey — APPROVED** [review]
Fixed conflicts myself as the stale PR was making me sad, happy to merge into lab only. Would love to upgrade it to a demo at some point, though.

**Comment by HurricanKai** [comment]
Just me or is GitHub refusing to merge this without a reason?

**Comment by Perksey** [comment]
Nah, the conversations must be resolved. It's a new requirement they've added, and I enabled it as I figured it'd make us look at all the comments again to make sure they're actually resolved. I'll resolve them real quick as this is the lab and we don't care.

**Pull request merged**
